### DISCUSSION TITLE
Feature request: export pattern data as plain-text JSON file format or similar

### DISCUSSION THREAD

**Discussion by fcento100** [WerWolv/ImHex#998] [Feedback]
It would be extremely useful if the pattern data could be exported to plain-text JSON file format for example. 

At the moment I'm having to re-implement what I've done in the pattern editor to deserialize the binary file in python. Happy to hear about alternative ways of achieving the same result.

**Comment by WerWolv** [comment]
Hey! This is possible to do right now through the `plcli` util. It's not properly released yet but you can download it from the Actions tab of https://github.com/WerWolv/PatternLanguage. The goal is to ultimately integrate this functionality into ImHex as well but we're not quite there yet

**Comment by mkpoli** [comment]
Just a quick note for anyone coming across this discussion later in case you missed it [like me](https://github.com/WerWolv/ImHex/discussions/2202):

There had already been a button shown as the following that lets you export to JSON and other formats easily. 

![image](https://github.com/user-attachments/assets/63f1967a-68dc-4a94-b02b-6bd2a30d4b29)

Thanks again to the developers — things have improved a lot since this thread was first opened 🙌
